client/src/planwise/client/scenarios/handlers.cljs

```diff
@@ -32,6 +32,40 @@
  (fn [db [_ path value]]
    (assoc-in db path value)))
 
+;; Views stack
+
+(rf/reg-event-db
+ :scenarios/push-view
+ in-scenarios
+ (fn [db [_ view]]
+   (println [".......x" view (:views-stack db)])
+   (update-in db [:views-stack] #(conj % view))))
+
+(rf/reg-event-db
+ :scenarios/pop-view
+ in-scenarios
+ (fn [db [_]]
+   (println [".......z" (:views-stack db)])
+   (update-in db [:views-stack] #(if (empty? %) % (pop %)))))
+
+(rf/reg-event-db
+ :scenarios/reset-state
+ in-scenarios
+ (fn [db [_]]
+   (assoc-in db [:views-stack] [])))
+
+(defn peek-view
+  [{:keys [views-stack]}]
+  (if (empty? views-stack)
+    :current-scenario
+    (peek views-stack)))
+
+(defn peek-view-1
+  [{:keys [views-stack]}]
+  (if (< (count views-stack) 2)
+    :current-scenario
+    (peek (pop views-stack))))
```

Comment:
This function is not used, correct?

Reply:
No, it isn't used anymore. I needed in some case but it didn't work.